src/coreclr/jit/codegencommon.cpp

```diff
@@ -4794,14 +4794,41 @@ void CodeGen::genCheckUseBlockInit()
                 }
             }
 
-            /* With compInitMem, all untracked vars will have to be init'ed */
-            /* VSW 102460 - Do not force initialization of compiler generated temps,
-                unless they are untracked GC type or structs that contain GC pointers */
-            CLANG_FORMAT_COMMENT_ANCHOR;
+            bool mustInitThisVar = false;
+
+            const bool isTemp      = varDsc->lvIsTemp;
+            const bool hasGCPtr    = varDsc->HasGCPtr();
+            const bool isTracked   = varDsc->lvTracked;
+            const bool isStruct    = varTypeIsStruct(varDsc);
+            const bool compInitMem = compiler->info.compInitMem;
 
-            if ((!varDsc->lvTracked || (varDsc->lvType == TYP_STRUCT)) && varDsc->lvOnFrame)
+            if (hasGCPtr && !isTracked)
+            {
+                JITDUMP("must init V%02u because it has a GC ref\n", varNum);
+                mustInitThisVar = true;
+            }
+            else if (hasGCPtr && isStruct)
+            {
+                // TODO-1stClassStructs: support precise liveness reporting for such structs.
+                JITDUMP("must init a tracked V%02u because it a struct with a GC ref\n", varNum);
+                mustInitThisVar = true;
+            }
+            else
             {
+                // We are done with tracked or GC vars, now look at untracked vars without GC refs.
+                if (!isTracked)
+                {
+                    assert(!hasGCPtr);
+                    if (compInitMem && !isTemp)
```

Comment:
Seems like we'll never see temps here due to `if (varDsc->lvIsTemp && !varDsc->HasGCPtr())` above?